cmd/minishift/cmd/update.go

```diff
@@ -0,0 +1,88 @@
+/*
+Copyright (C) 2017 Red Hat, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cmd
+
+import (
+	"fmt"
+	//"strings"
+
+	"github.com/minishift/minishift/pkg/minishift/update"
+	"github.com/minishift/minishift/pkg/util/os/atexit"
+
+	"github.com/minishift/minishift/pkg/util"
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+	"strings"
+)
+
+const (
+	updateHttpProxy  = "http-proxy"
+	updateHttpsProxy = "https-proxy"
+)
+
+var updateCmd = &cobra.Command{
+	Use:   "update",
+	Short: "Update to latest version of Minishift.",
+	Long:  `Checks for the latest version of Minishift, prompt the user and update the binary if user answers with 'y'.`,
+	Run:   runUpdate,
+}
+
+func runUpdate(cmd *cobra.Command, args []string) {
+	proxyConfig := handleProxies()
+	proxyConfig, err := util.NewProxyConfig(viper.GetString(httpProxy), viper.GetString(httpsProxy), viper.GetString(noProxyList))
```

Comment:
This is not quite right, you introduced _updateHttpProxy_ and _updatesHttpProxy_, but you are here referencing the constants from start.go. Also, you should not be using no proxy constant. I'll take care of this.